lib/webmock/http_lib_adapters/net_http.rb

```diff
@@ -128,7 +128,11 @@ def start_without_connect
 
 
         def ensure_actual_connection
-          do_start if @socket.is_a?(StubSocket)
+          if @socket.is_a?(StubSocket)
+            @socket&.close
+            @socket = nil
```

Comment:
Why are these necessary? How is it better simulating net/http?

It causes issues when running tests with tools like https://github.com/shopify/toxiproxy where the test involves starting a successful HTTP connection to a local server, and then mimics that server failing using toxiproxy, and attempting another request. The exception raised is "undefined method `closed?' for nil", which is coming from https://github.com/ruby/net-http/blob/d39f1e35da0094759356a49a3c96fea84325c388/lib/net/http.rb#L2431.